lib/converters/v1.0.0/converter-v1-to-v21.js

```diff
@@ -0,0 +1,153 @@
+var inherits = require('inherits'),
+    _ = require('lodash').noConflict(),
+
+    url = require('../../url'),
+    constants = require('../../constants'),
+    BaseBuilders = require('./converter-v1-to-v2').Builders,
+    Builders,
+    typeMap = {
+        string: 'string',
+        boolean: 'boolean',
+        number: 'number'
+    };
+
+inherits(Builders = function ConversionBuilder () {
+    Builders.super_.apply(this, arguments);
+}, BaseBuilders);
+
+_.assign(Builders.prototype, {
+
+    /**
+     * Derives v2.1.0 collection info from a v1.0.0 collection object.
+     *
+     * @param collectionV1 - The v1.0.0 collection object to be converted to v2.1.0.
+     * @return {Object} - The compiled v2.x collection info manifest.
+     */
+    info: function (collectionV1) {
+        var info = Builders.super_.prototype.info.call(this, collectionV1);
+
+        info.schema = constants.SCHEMA_V2_1_0_URL;
+
+        return info;
+    },
+
+    /**
+      * Converts collection request urls from v1.0.0 to v2.1.0
+      *
+      * @param {Object} requestV1 - The v1.0.0 request url to be converted to v2.1.0.
+      * @return {Object} - The objectified v2.1.0 compliant URL.
+      */
+    url: function (requestV1) {
+        var v21Url = Builders.super_.prototype.url.call(this, requestV1);
+
+        return _.isString(v21Url) ? url.parse(v21Url) : v21Url;
+    },
+
+    /**
+     * A number of auth parameter names have changed from V1 to V2. This function calls the appropriate
+     * mapper function, and creates the V2 auth parameter object.
+     *
+     * @param {Object} requestV1 - A Collection V1 compliant request instance
+     * @returns {{type: *}} - The v2.1.0 compliant request object
+     */
+    auth: function (requestV1) {
+        var result,
+            auth = Builders.super_.prototype.auth.call(this, requestV1);
+
+        if (!(auth && auth.type && auth[auth.type])) { return; }
+
+        result = { type: auth.type }; // @todo: Replace this with ES6 style dynamic property definition later
+        result[auth.type] = _.map(auth[auth.type], function (value, key) {
+            return {
+                key: key,
+                value: value,
+                type: typeMap[typeof value] || 'any'
```

Comment:
if a type is already present anywhere, respect that.

Reply:
That is already being done, this is a v1 to v2.1 conversion. Type information is not available for v1 formats, so a derivation will have to be done here.

Reply:
Addressed